controllers/manager/factory_install_info.go

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright(c) 2024 Wind River Systems, Inc. */
+
+package manager
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	perrors "github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConfigMap to record and drive the configuration after factory install
+// Sample configMap:
+/*
+apiVersion: v1
+kind: ConfigMap
+metadata:
+	name: factory-install
+	namespace: deployment
+data:
+	factory-installed: "false"  # Initial state (change to "true" if factory-installed)
+	factory-config-finalized: "false" # Initial state (change to "true" if configuration is complete)
+	system-abcd-default-updated: "false" # Initial state (change to "true" if resource default updated)
+*/
+const FactoryInstallConfigMapName = "factory-install"
+
+func (m *PlatformManager) GetFactoryInstall(ns string) (bool, error) {
```

Comment:
Understood, does it make sense to rename the function to `IsFactoryInstallExpected`?
Or maybe a comment regarding the return values would be helpful.

Reply:
make sense, I'm going to add a comment against this method